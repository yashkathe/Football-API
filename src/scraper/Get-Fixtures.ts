// import axios from "axios";
// import cheerio from "cheerio";

// import { competitionCode } from "../endpoints/variable-codes";
// import { endpoints } from "../endpoints/endpoints";
// import type { isFixture } from "../types/types";

// export const getFixtures = (): Promise<isFixture[]> => {
//     try {
//         const endpoint: string = `${endpoints.competetion}/${competitionCode.premierLeague}/fixtures`;
//         let fixtures: isFixture[] = [];

//         return new Promise(async (resolve, reject) => {
//             const response = await axios(endpoint);
//             const $ = cheerio.load(response.data);

//             $("article.simple-match-card").each(function () {
//                 const homeTeam: string = $(this)
//                     .find("div > div.simple-match-card__teams-content > of-simple-match-card-team:nth-child(1) > div > span.title-8-medium.simple-match-card-team__name")
//                     .text();
//                 console.log(homeTeam);
//             });
//         });
//     } catch (error: any) {
//         throw new Error(error);
//     }
// };
