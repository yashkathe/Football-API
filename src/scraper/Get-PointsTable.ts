import axios from "axios";
import cheerio from "cheerio";

import { competitionCode } from "../endpoints/variable-codes";
import { endpoints } from "../endpoints/endpoints";
import type { isTeamInPointsTable } from "../types/types";

const getPointsTable = (competitionCode: string): Promise<isTeamInPointsTable[]> => {
    try {
        const endpoint: string = `${endpoints.competetion}/${competitionCode}/table`;
        let pointsTable: isTeamInPointsTable[] = [];
        //
        return new Promise(async (resolve, reject) => {
            const response = await axios(endpoint);
            const $ = cheerio.load(response.data);
            //
            $("li.standings__row").each(function () {
                const position: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(1) > span:nth-child(1)").text());
                const team: string = $(this).find("a:nth-child(1) > div:nth-child(3) > p:nth-child(2)").text().trim();
                const gamesPlayed: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(4) > span:nth-child(1)").text());
                const gamesWon: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(5) > span:nth-child(1)").text());
                const gamesLost: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(7) > span:nth-child(1)").text());
                const gamesDrawn: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(6) > span:nth-child(1)").text());
                const goalDifference: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(8) > span:nth-child(1)").text());
                const points: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(9) > span:nth-child(1)").text());

                if (
                    !Number.isNaN(position) &&
                    team.length !== 0 &&
                    !Number.isNaN(gamesPlayed) &&
                    !Number.isNaN(gamesWon) &&
                    !Number.isNaN(gamesLost) &&
                    !Number.isNaN(gamesDrawn) &&
                    !Number.isNaN(goalDifference) &&
                    !Number.isNaN(points)
                ) {
                    const teamInPointsTable: isTeamInPointsTable = {
                        position,
                        team,
                        gamesPlayed,
                        gamesWon,
                        gamesLost,
                        gamesDrawn,
                        goalDifference,
                        points,
                    };
                    pointsTable.push(teamInPointsTable);
                }
            });
            resolve(pointsTable);
        });
    } catch (error: any) {
        throw new Error(error);
    }
};

export const premierLeagueStandings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.premierLeague);
};

export const laLigaStandings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.laliga);
};

export const serieAStandings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.serieA);
};

export const ligue1Standings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.ligue1);
};

export const bundesligaStandings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.bundesliga);
};
