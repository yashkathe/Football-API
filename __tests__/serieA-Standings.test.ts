import { serieAStandings } from "../src/scraper/Get-PointsTable";

test("serieA standings", async () => {
    expect(await serieAStandings()).toMatchSnapshot();
});
