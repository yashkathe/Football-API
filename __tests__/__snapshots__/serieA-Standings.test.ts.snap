// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serieA standings 1`] = `
[
  {
    "gamesDrawn": 2,
    "gamesLost": 0,
    "gamesPlayed": 15,
    "gamesWon": 13,
    "goalDifference": 25,
    "points": 41,
    "position": 1,
    "team": "Napoli",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 2,
    "gamesPlayed": 15,
    "gamesWon": 10,
    "goalDifference": 14,
    "points": 33,
    "position": 2,
    "team": "Milan",
  },
  {
    "gamesDrawn": 4,
    "gamesLost": 2,
    "gamesPlayed": 15,
    "gamesWon": 9,
    "goalDifference": 17,
    "points": 31,
    "position": 3,
    "team": "Juventus",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 3,
    "gamesPlayed": 15,
    "gamesWon": 9,
    "goalDifference": 15,
    "points": 30,
    "position": 4,
    "team": "Lazio",
  },
  {
    "gamesDrawn": 0,
    "gamesLost": 5,
    "gamesPlayed": 15,
    "gamesWon": 10,
    "goalDifference": 12,
    "points": 30,
    "position": 5,
    "team": "Inter",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 4,
    "gamesPlayed": 15,
    "gamesWon": 8,
    "goalDifference": 7,
    "points": 27,
    "position": 6,
    "team": "Atalanta",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 4,
    "gamesPlayed": 15,
    "gamesWon": 8,
    "goalDifference": 4,
    "points": 27,
    "position": 7,
    "team": "Roma",
  },
  {
    "gamesDrawn": 6,
    "gamesLost": 3,
    "gamesPlayed": 15,
    "gamesWon": 6,
    "goalDifference": 7,
    "points": 24,
    "position": 8,
    "team": "Udinese",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 6,
    "goalDifference": -1,
    "points": 21,
    "position": 9,
    "team": "Torino",
  },
  {
    "gamesDrawn": 4,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 5,
    "goalDifference": -2,
    "points": 19,
    "position": 10,
    "team": "Fiorentina",
  },
  {
    "gamesDrawn": 4,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 5,
    "goalDifference": -5,
    "points": 19,
    "position": 11,
    "team": "Bologna",
  },
  {
    "gamesDrawn": 5,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 4,
    "goalDifference": -5,
    "points": 17,
    "position": 12,
    "team": "Salernitana",
  },
  {
    "gamesDrawn": 5,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 4,
    "goalDifference": -7,
    "points": 17,
    "position": 13,
    "team": "Empoli",
  },
  {
    "gamesDrawn": 1,
    "gamesLost": 9,
    "gamesPlayed": 15,
    "gamesWon": 5,
    "goalDifference": -6,
    "points": 16,
    "position": 14,
    "team": "Monza",
  },
  {
    "gamesDrawn": 4,
    "gamesLost": 7,
    "gamesPlayed": 15,
    "gamesWon": 4,
    "goalDifference": -7,
    "points": 16,
    "position": 15,
    "team": "Sassuolo",
  },
  {
    "gamesDrawn": 6,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 3,
    "goalDifference": -3,
    "points": 15,
    "position": 16,
    "team": "Lecce",
  },
  {
    "gamesDrawn": 4,
    "gamesLost": 8,
    "gamesPlayed": 15,
    "gamesWon": 3,
    "goalDifference": -12,
    "points": 13,
    "position": 17,
    "team": "Spezia",
  },
  {
    "gamesDrawn": 7,
    "gamesLost": 8,
    "gamesPlayed": 15,
    "gamesWon": 0,
    "goalDifference": -15,
    "points": 7,
    "position": 18,
    "team": "Cremonese",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 11,
    "gamesPlayed": 15,
    "gamesWon": 1,
    "goalDifference": -21,
    "points": 6,
    "position": 19,
    "team": "Sampdoria",
  },
  {
    "gamesDrawn": 2,
    "gamesLost": 12,
    "gamesPlayed": 15,
    "gamesWon": 1,
    "goalDifference": -17,
    "points": 5,
    "position": 20,
    "team": "Hellas Verona",
  },
]
`;
